### PR TITLE
feature/200_remove-temp-clustering-switch

### PR DESCRIPTION
**Pull request by cschwinderg**
## Related Issues:
* https://jira.epa.gov/browse/HMW-200

## Main Changes:
* Removed the Temp Monitoring Clustering switch and updated the code so that clustering is automatically activated if there are more than 20 stations in the HUC.

## Steps To Test:
1. Navigate to http://localhost:3000/community/portland%20or/monitoring
2. Verify the Past Water Conditions layer is being clustered
3. Verify the `TEMP Monitoring Clustering` switch is no longer in the app
4. Navigate to http://localhost:3000/community/columbia%20sc/monitoring
5. Verify the Past Water Conditions layer is not being clustered
6. Verify the `TEMP Monitoring Clustering` switch is no longer in the app

### CODE REVIEW

**Review by maxdiebold-erg — CHANGES_REQUESTED** [review]
@cschwinderg The clustering looks good, I think you picked a good radius so the clusters don't overlap. However, the highlighting isn't working for me when hovering over an accordion item.

I suspect this is because the clusters are different graphics. What you would need to do, I think, is query the visible graphics with `FeatureLayerView.queryFeatures()` to get the cluster graphics, then get the underlying features associated with each graphic, by setting the graphic's ID in the layerView query's aggregateIds property and querying each graphic.

Or, there might be a way to query the visible graphics by an underlying feature ID, but I can't seem to find where the feature data is stored in the FeatureReductionCluster object.

File: app/client/src/components/shared/LocationMap.js [hunk 1]
```diff
@@ -1183,6 +1190,58 @@ function LocationMap({ layout = 'narrow', windowHeight, children }: Props) {
...
     });
 
+    if (stationsSorted.length > 20) {
+      monitoringLocationsLayer.featureReduction = {
+        type: 'cluster',
+        clusterRadius: '100px',
+        clusterMinSize: '24px',
+        clusterMaxSize: '60px',
+        popupEnabled: true,
+        popupTemplate: {
```
[the remark this answers is not in the document]
Comment (cschwinderg) [reply app/client/src/components/shared/LocationMap.js:1200]: Those action buttons are the default buttons for cluster popups. I don't see any built in ways to default the popup to the list in the ArcGIS JS API. We could do something hacky where we click the button through code, but I think it is better to just stick with default behavior of the popup.

**Comment by cschwinderg** [comment]
@maxdiebold-erg Thanks for noticing the highlighting issue. I think I'll create a separate ticket/PR for that, since that could end up being a complicated issue. I'll leave this PR as just being about making clustering enabled when a HUC has more than 20 stations.

**Comment by cschwinderg** [comment]
@maxdiebold-erg I don't define that. Those items are in there by default in the ArcGIS JS API and the API doesn't provide an easy way to default to that list view.

**Comment by cschwinderg** [comment]
@maxdiebold-erg I've been researching the highlighting issue. Unfortunately, the `FeatureLayerView.queryFeatures()` doesn't return the cluster graphics. I haven't been able to find any other way to query the clusters or even figure out if the graphic is in a cluster. 

The API only seems to allow you to find out what features are in a cluster if you click on a cluster. I'm going to post a question on the Esri forums about this. 

I'm going to go ahead and merge this PR in though, since the highlighting and possible map popup changes will be separate tickets/PRs.

**Pull request merged**